### DISCUSSION TITLE
Better and more convenient extension development

### DISCUSSION THREAD

**Discussion by zmh-program** [TurboWarp/extensions#597] [Show and tell]
I like making turbowarp extensions, and I've made the cloud database extension and the OpenAI extension before. 

But I have found a problem, especially when the project gets large, that it can be a bit time consuming to weave between block parameters and functions, and I think this may not be as well designed. In particular, the readability becomes worse at post-maintenance time.

So I'm thinking about how to make the extension interface more reasonable and make extension production easier and more efficient, instead of spending a lot of time or even sacrificing readability because of a configuration. 

So I wrote a plugin [turbowarp-clipx](https://github.com/zmh-program/turbowarp-clipx). at first, although I think it has been able to reduce a lot of time wasted because of the configuration, but I think it is still not worth to promote. Later, I wrote a built-in cache, a better interface style, and I finally figured out how to automate the translation.

Like many front-end projects, the i18n process is tedious and time-consuming, especially when it comes to changes, and

Automated translation cuts the time spent on manual translation the most.

So I made it. In the end, it turned out to be less than a quarter of the total code in many cases, and most importantly, the developer could develop extensions very efficiently, very easily, and with very simple continuous CI/CD, without having to care about shuttling between configurations and methods.
As well as I also developed the function of github workflow to automatically package and publish.

So, how should I do? It's very simple, just use the template, see the project for details :)

**Comment by LilyMakesThings** [comment]
Did you seriously upvote this with alts 💀

**Reply by zmh-program** [reply]
Sorry, I'm not sure what "alts" means here, could you explain it?

**Reply by yuri-kiss** [reply]
alternate accounts or accounts owned and or operated by you, which in this case were used for the social action of "upvoting" and adding reactions to your post.

**Reply by zmh-program** [reply]
?? I am not sure where you are discussing.

**Reply by zmh-program** [reply]
🤔So here's my alternate account? 
@SurvExe1Pc 
![Screenshot_20230618_091445_edit_212401673787232](https://github.com/TurboWarp/extensions/assets/112773885/29170f23-2de8-4214-8200-96d0861e0b43)
![Screenshot_20230618_091454_edit_212390149735394](https://github.com/TurboWarp/extensions/assets/112773885/b07deb7a-b392-4ca5-9a7b-7debf2267452)

**Comment by yuri-kiss** [comment]
> 🤔So here's my alternate account? @SurvExe1Pc ![Screenshot_20230618_091445_edit_212401673787232](https://user-images.githubusercontent.com/112773885/246638610-29170f23-2de8-4214-8200-96d0861e0b43.jpg) ![Screenshot_20230618_091454_edit_212390149735394](https://user-images.githubusercontent.com/112773885/246638614-b07deb7a-b392-4ca5-9a7b-7debf2267452.jpg)

https://github.com/TurboWarp/extensions/discussions/597#discussion-5309637
you used your alts to upvote and react to the post ^^

**Reply by yuri-kiss** [reply]
![image](https://github.com/TurboWarp/extensions/assets/135030944/ffda1221-c3eb-459d-ada1-5124a6c6f9aa)

**Reply by yuri-kiss** [reply]
![image](https://github.com/TurboWarp/extensions/assets/135030944/5c792f7a-12fa-4a39-9312-cd0d789645cc)

**Reply by zmh-program** [reply]
and then? 🤔

**Reply by yuri-kiss** [reply]
thats it, other than the commets :|

**Reply by zmh-program** [reply]
And I don't think that's the direction of the discussion. 🤔

**Reply by yuri-kiss** [reply]
that was and is Lol, based on the comment by @LilyMakesThings https://github.com/TurboWarp/extensions/discussions/597#discussioncomment-6206935

**Reply by zmh-program** [reply]
So I wonder what is the meaning of discussing this at such great length😊

**Reply by yuri-kiss** [reply]
to answer "https://github.com/TurboWarp/extensions/discussions/597#discussioncomment-6206942"

**Reply by yuri-kiss** [reply]
and in reply to ["so heres my alt acc post"](https://github.com/TurboWarp/extensions/discussions/597#discussioncomment-6206969): no the github account

**Reply by zmh-program** [reply]
OK.

**Comment by zmh-program** [comment]
> and in reply to ["so heres my alt acc post"](https://github.com/TurboWarp/extensions/discussions/597#discussioncomment-6206969): no the github account

?

**Reply by yuri-kiss** [reply]
> > and in reply to ["so heres my alt acc post"](https://github.com/TurboWarp/extensions/discussions/597#discussioncomment-6206969): no the github account
> 
> ?

^ exactly

**Reply by zmh-program** [reply]
^ https://github.com/zmh-program/turbowarp-clipx/stargazers

**Reply by yuri-kiss** [reply]
2 are you in the reactions on some

**Reply by zmh-program** [reply]
So what do you think of my project and what do you suggest?

**Reply by yuri-kiss** [reply]
i cant read Chinese,  but maybe you could make a simple web UI or smth idk im not good at suggestions.

**Reply by zmh-program** [reply]
![Screenshot_20230618_100030_edit_215159031298259](https://github.com/TurboWarp/extensions/assets/112773885/7511d016-ca74-4edc-bc12-5fa175631df5)
😊
[English Readme](https://github.com/zmh-program/turbowarp-clipx/blob/main/docs/en.md)

**Reply by yuri-kiss** [reply]
Oml i am blind

**Reply by zmh-program** [reply]
thanks

**Reply by zmh-program** [reply]
Thanks I'm also thinking about how to embed blocky or webui😀
